api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -348,6 +348,10 @@ message HttpConnectionManager {
     repeated HttpFilter filters = 2;
   };
   repeated UpgradeConfig upgrade_configs = 23;
+
+  // If true, the order of encoder filters will be reversed to that of filters
+  // configured in the HTTP filter chain.
+  bool reverse_encoders_order = 27;
```

Comment:
what is the use case? it would be good to have some explanation associated with such settings. Otherwise, we end up having a plethora of "invert_x", "reverse_y" boolean options all over the config

Reply:
This is a bug fixing in envoy. The design of envoy was to have reserved order of encoder filters to the configuration in the filter chain. However, the current implementation in envoy is not following this order. In order to avoid breaking anyone who depends on this bug, we decided to fix this as an opt-in behavior.

For more details, see https://github.com/envoyproxy/envoy/issues/4599

Reply:
Does it make sense to link https://github.com/envoyproxy/envoy/issues/4599 in the comment for this? It adds valuable context.

Reply:
I don't think we have to go to this extreme to preserve backward compatibility for incorrect behavior. It seems counter intuitive, given that the goal of the API is to describe the system (not workarounds for "correct" behavior). Put another way, if those other systems depended on the incorrect behavior, then those systems are also not functioning correctly are they?

Reply:
As @htuch suggested, we will mark this field deprecated and put in the release note too.